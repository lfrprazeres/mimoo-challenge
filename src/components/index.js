import Button from './Button';
import Products from './Products';

export {
    Button,
    Products
}