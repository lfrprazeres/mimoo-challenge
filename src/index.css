@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;500;600;700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@100;400;500;600;700&display=swap');

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}

html,body {
  min-width: 100%;
  min-height: 100%;
  overflow-x: hidden;
}

a {
  text-decoration: none;
}

li {
  list-style-type: none;
}