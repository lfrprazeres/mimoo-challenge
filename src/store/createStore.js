
import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import Saga from '../saga';
import { Reducers } from '../reducers';
import { composeWithDevTools } from 'redux-devtools-extension';
import { apiMiddleware } from 'redux-api-middleware';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

const persistConfig = {
    key: 'root',
    storage,
}

const persistedReducer = persistReducer(persistConfig, Reducers);

const sagaMiddleware = createSagaMiddleware();

let store = createStore(
    persistedReducer,
    composeWithDevTools(
        applyMiddleware(
            sagaMiddleware,
            apiMiddleware
        )
    )
);

sagaMiddleware.run(Saga);

const persistor = persistStore(store);
export { store, persistor }