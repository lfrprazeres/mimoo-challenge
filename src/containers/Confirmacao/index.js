import React, { useState } from 'react';
import { Button } from '../../components';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { saveNewProduct } from '../../actions/user';
import { device } from '../../utils/deviceSizes';

const ConfirmacaoStyled = styled.main`
    align-items: center;
    background-color: ${props => props.lightBrown};
    display: flex;
    flex-direction: column;
    padding-top: 43px;
    padding-bottom: 50px;
    position: relative;
    min-height: 100vh;
    @media ${device.mobileL} {
        align-items: center;
        flex-direction: column;
        height: 100vh;
        justify-content: center;
    }
    .identifier {
        color: white;
        margin-left: -5px;
        h4 {
            font-size: 25px;
            font-weight: 500;
            margin-bottom: 4px;
            text-align: center;
            width: 242px;
        }
        span {
            font-size: 34px;
            font-weight: bold
        }
    }
    .image {
        margin-left: 20px;
        width: 265px;
        img {
            width: 100%;
        }
    }
    .congrats {
        color: ${props => props.black};
        display: flex;
        flex-direction: column;
        span {
            font-size: 20px;
            text-align: center;
            &:first-child {
                font-weight: bold;
                margin: -20px 0 13px -6px;
            }
            &:nth-child(2) {
                font-weight: 500;
                margin-left: -7px;
            }
        }
    }
    div {
        &:last-child {
            @media ${device.mobileS} {
                bottom: 0px;
            }
        }
    }
`;

function Confirmacao(props) {
    const [ pointsEarned ] = useState(100);
    const {
        colors,
        product = {},
        onSave
    } = props;
    return (
        <ConfirmacaoStyled lightBrown={colors.lightBrown} black={colors.black}>
            <div className="identifier">
                <h4> Identificamos <br /> que você consome </h4>
                <span> {product.name} </span>
            </div>
            <div className="image">
                <img alt={product.name} src={product.image} />
            </div>
            <div className="congrats">
                <span>
                    Parabéns <br /> Você ganhou {pointsEarned} pontos!
                </span>
                <span>
                    Continue para ganhar ainda <br /> mais pontos
                </span>
            </div>
            <Button to="/home" label="Salvar" bg="white" color={colors.green} onClick={() => onSave(product, pointsEarned)} />
        </ConfirmacaoStyled>
    )
}

const mapStateToProps = state => ({
  colors: state.app.defaultColors,
  product: state.user.newProduct
});

const mapDispatchToProps = dispatch => {
    return {
        onSave: async (product, pointsEarned) => {
            await dispatch(saveNewProduct(product, pointsEarned));
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Confirmacao);