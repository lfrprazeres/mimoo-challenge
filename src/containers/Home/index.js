import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import StarRoundedIcon from '@material-ui/icons/StarRounded';
import { Products } from '../../components';
import { IconButton } from '@material-ui/core';
import AddRoundedIcon from '@material-ui/icons/AddRounded';
import { searchProduct } from '../../actions/user';
import { useHistory } from 'react-router-dom';

const HomeStyled = styled.main`
    margin: 30px 0 0 36px;
    min-height: 100vh;
    h1 {
        color: ${props => props.black};
        font-size: 34px;
        font-weight: 500;
        margin-bottom: 8px;
    };
    h4 {
        color: ${props => props.black};
        font-size: 16px;
        font-weight: 400;
        margin-bottom: 20px;
        width: 275px;
    };
    .points {
        color: ${props => props.brown};
        .pointsTitle {
            align-items: flex-start;
            display: flex;
            margin-bottom: 1px;
            svg {
                width: 19px;
                margin: 0 3px 0 -3px;
            }
            span {
                font-size: 18px;
                margin-top: -1px;
            }
        }
        .pointsValue {
            font-size: 25px;
            font-weight: bold;
            margin-left: 19px;
        }
    }
    .addButton {
        background-color: ${props => props.green};
        bottom: 49px;
        padding: 5px;
        position: fixed;
        right: 47px;
        transition: .2s ease;
        &:hover {
            background-color: ${props => props.green};
            filter: brightness(110%);
        }
        svg {
            color: white;
            height: 58px;
            width: 58px;
        }
    }
`;

function Home(props) {
    const history = useHistory();
    useEffect(() => {
        const {
            getProducts
        } = props;
        if(user.products.length < 1) {
            getProducts();
        }
    }, [])

    const {
        user,
        colors
    } = props;
    return(
        <HomeStyled black={colors.black} brown={colors.brown} green={colors.green}>
            <h1>Olá {user.name}!</h1>
            <h4>Adicione mais produtos à sua lista e ganhe pontos!</h4>
            <div className="points">
                <div className="pointsTitle">
                    <StarRoundedIcon />
                    <span>
                        Pontos
                    </span>
                </div>
                <span className="pointsValue"> {user.points} </span>
            </div>
            <Products/>
            <IconButton className="addButton" onClick={() => history.push("/scan")}>
                <AddRoundedIcon />
            </IconButton>
        </HomeStyled>
    )
}

const mapStateToProps = state => ({
    user: state.user,
    colors: state.app.defaultColors
});

const mapDispatchToProps = dispatch => {
    return {
        getProducts: async () => await dispatch(searchProduct())
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(React.memo(Home));