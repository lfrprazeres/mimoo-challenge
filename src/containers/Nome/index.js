import React, { useState, useEffect } from 'react';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { useHistory } from "react-router-dom"; 
import IconButton from '@material-ui/core/IconButton';
import { TextField } from '@material-ui/core';
import { Button } from '../../components';
import { createMuiTheme } from '@material-ui/core/styles';
import { ThemeProvider } from '@material-ui/styles';
import { login } from '../../actions/user';
import * as utilAnimation from '../../utils/animations/viewSlide';
import { device } from '../../utils/deviceSizes';

const theme = createMuiTheme({
    overrides: {
        MuiFormControl: {
            root: {
                margin: "24px 0 0 11px",
                width: "88%",
            }
        },
        MuiInputLabel: {
            root: {
                fontSize: 22,
                marginTop: "-4px",
                fontFamily: "'Poppins', sans-serif;",
                "&$focused": {
                    color: "#00C3A3"
                }
            }
        },
        MuiInput: {
            underline: {
                color: "#3A3A3A",
                fontSize: 22,
                '&:after': {
                    borderBottom: "1px solid #00C3A3"
                },
                '&:before': {
                    borderBottom: "1px solid #00C3A3"
                },
                "&&:hover:before": {
                    borderBottom: "1px solid #00C3A3"
                }
            }
        },
    }
});


const NomeStyled = styled.main`
    &.page-enter {
        animation: ${props => props.enter} 0.2s forwards;
    }
    &.page-exit {
        animation: ${props => props.exit} 0.2s forwards;
    }
    .content {
        align-items: flex-start;
        display: flex;
        flex-direction: column;
        height: 100vh;
        padding: 14px 0 80px 24px;
    }
    .arrowIcon {
        color: ${props => props.green};
        font-weight: bold;
        width: 35px;
        height: 35px;
    }
    h1 {
        color: ${props => props.black};
        font-size: 29px;
        font-weight: 500;
        letter-spacing: .60px;
        margin-top: 41px;
        margin-left: 12px;
    }
`

function Nome(props) {
    const [name, setName] = useState("");
    const {
        colors,
        addName,
    } = props;
    const {
        animation = {}
    } = props.location;
    const history = useHistory();
    return (
        <NomeStyled green={colors.green} black={colors.black} enter={animation.enter} exit={animation.exit}>
            <div className="content">
                <IconButton aria-label="back" onClick={() => history.goBack()}>
                    <ArrowBackIcon className="arrowIcon"/>
                </IconButton>
                <h1> Informe seu nome </h1>
                <ThemeProvider theme={theme}>
                    <TextField label="Nome" value={name} onChange={(e) => setName(e.target.value)}/>
                </ThemeProvider>
            </div>
            <Button
                to="/home"
                label="Continuar"
                onClick={() => addName(name)} bg={colors.green}
                color="white"
                animation={{
                    enter: utilAnimation.slideInRight,
                    exit: utilAnimation.slideOutRight
                }}
            />
        </NomeStyled>
    )
}

const mapStateToProps = state => ({
  colors: state.app.defaultColors
});

const mapDispatchToProps = dispatch => {
    return {
        addName: async (name) => {
            await dispatch(login(name))
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Nome);