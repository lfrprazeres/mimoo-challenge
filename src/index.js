import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import Routes from './routes';
import { Provider } from 'react-redux';
import { store, persistor } from './store/createStore';
import { PersistGate } from 'redux-persist/integration/react';

ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Routes />
      </PersistGate>
    </Provider>
  </React.StrictMode>,
  document.getElementById('root')
);
