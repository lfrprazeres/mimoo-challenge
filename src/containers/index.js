import SplashScreen from './SplashScreen';
import Nome from './Nome';
import Home from './Home';
import Scan from './Scan';
import Confirmacao from './Confirmacao';

export {
    SplashScreen,
    Nome,
    Home,
    Scan,
    Confirmacao
}