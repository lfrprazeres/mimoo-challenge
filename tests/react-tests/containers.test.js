import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import { store } from '../../src/store/createStore';
import { SplashScreen, Nome, Home, Scan, Confirmacao } from '../../src/containers';

configure({adapter: new Adapter()});

describe('Container tests', () => {
    it('should render SplashScreen without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <SplashScreen />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render Nome without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Nome />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render Home without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Home />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render Scan without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Scan />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should render Confirmacao without crash with the properly props', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <Confirmacao />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot(); 
    });
});