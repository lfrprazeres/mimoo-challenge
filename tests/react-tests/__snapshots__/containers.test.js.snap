// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container tests should render Confirmacao without crash with the properly props 1`] = `ShallowWrapper {}`;

exports[`Container tests should render Home without crash with the properly props 1`] = `ShallowWrapper {}`;

exports[`Container tests should render Nome without crash with the properly props 1`] = `ShallowWrapper {}`;

exports[`Container tests should render Scan without crash with the properly props 1`] = `ShallowWrapper {}`;

exports[`Container tests should render SplashScreen without crash with the properly props 1`] = `ShallowWrapper {}`;
